backend/src/transaction_event/protocol_level_tokens.rs

```diff
@@ -522,26 +525,38 @@ impl From<TokenUpdate> for PreparedTokenUpdate {
                     "unpause" => Some(TokenUpdateModuleType::Unpause),
                     _ => None,
                 };
-                TokenUpdateEventType::TokenModule
+                (TokenUpdateEventType::TokenModule, 0u64, 0i32)
             }
             TokenEventDetails::Transfer(e) => {
                 from = Some(e.from.address.to_string());
                 to = Some(e.to.address.to_string());
                 plt_amount_change =
                     BigDecimal::from(e.amount.value.parse::<f64>().unwrap_or(0.0) as u64);
-                TokenUpdateEventType::Transfer
+                (
+                    TokenUpdateEventType::Transfer,
+                    e.amount.value.parse::<u64>().unwrap_or(0),
+                    e.amount.decimals.parse::<i32>().unwrap_or(0),
+                )
             }
             TokenEventDetails::Mint(e) => {
                 target = Some(e.target.address.to_string());
                 plt_amount_change =
                     BigDecimal::from(e.amount.value.parse::<f64>().unwrap_or(0.0) as u64);
-                TokenUpdateEventType::Mint
+                (
+                    TokenUpdateEventType::Mint,
+                    e.amount.value.parse::<u64>().unwrap_or(0),
```

Comment:
I wonder if we should throw an error in all these cases where parsing fails as we should be able to parse the  value (or we have a bug in the code).

Reply:
I think it is a good idea to throw error.